include/logger/sink.hpp

```diff
@@ -123,17 +124,19 @@ bool logger::DefaultSink::writeToFile(std::string formatted_data)
 {
   std::fstream fout;
   fout.open(log_file_name, std::ios::app);
+
   if (fout.is_open()) 
   {
-	//considering corner case.if file is empty append from begining , else append from next line.
-    fout.seekg(0, std::ios::end);
-    auto length = fout.tellg();
-    if (length != 0) {fout << "\n";}
-    fout << formatted_data;
+    fout<<formatted_data<<std::endl;
     fout.close();
+    return true;
+  }
+  else{
+ // To be updated later when error handling will be dealt .
+  throw std::exception("File cannot be opened!");
```

Comment:
decides whether you want to return and error or use an exceptions. donot mix both.